src/core/thread/mesh_forwarder.cpp

```diff
@@ -1256,6 +1255,12 @@ void MeshForwarder::UpdateSendMessage(Error aFrameTxError, Mac::Address &aMacDes
         break;
 
     case Message::kSubTypeMleChildIdRequest:
+        if (mSendMessage->GetTxSuccess() && !Get<Mle::Mle>().IsRxOnWhenIdle())
+        {
+            Get<DataPollSender>().SetAttachMode(true);
+            Get<MeshForwarder>().SetRxOnWhenIdle(false);
```

Comment:
We are in `MeshForwarder` class already :)
```suggestion
            SetRxOnWhenIdle(false);
```

Reply:
Thanks, fixed it.